contracts/RLPReader.sol

```diff
@@ -29,6 +55,19 @@ library RLPReader {
         return RLPItem(item.length, memPtr);
     }
 
+    /*
+    * @dev Create an iterator.
+    * @param self The RLP item.
+    * @return An 'Iterator' over the item.
+    */
+    function iterator(RLPItem memory self) internal pure returns (Iterator memory it) {
+        if (!isList(self))
```

Comment:
can we make this `require(isList(self))`. Error strings cost a lot of extra gas

Reply:
Yep, changed it accordingly in the latest commit. Also, removed the revert message in the `next` method.